add-schema.sh

```diff
@@ -38,69 +38,102 @@ fi
-  git submodule add -b ${gitBranch} ${gitRepository} schemas/${schema}
+  git submodule add --force -b ${gitBranch} ${gitRepository} schemas/${schema}
 fi
 
 
 # Add schema module in schemas/pom.xml
 line=$(grep -n ${schema} schemas/pom.xml | cut -d: -f1)
 
-if [ ! $line ]
+if [ ! -n "$line" ]
 then
-  line=$(grep -n 'iso19139</module>' schemas/pom.xml | cut -d: -f1)
+  line=$(grep -n '</profiles>' schemas/pom.xml | cut -d: -f1)
+  insertLine=$(($line - 1))
 
   echo "Adding schema ${schema} to schemas/pom.xml"
 
   sed $sedopt -f /dev/stdin schemas/pom.xml << SED_SCRIPT
-  ${line} a\\
-\    <module>${schema}</module>
+  ${insertLine} a\\
+\    <profile>\\
+\      <id>schema-${schema}</id>\\
+\      <activation>\\
+\        <file><exists>${schema}</exists></file>\\
+\      </activation>\\
+\      <modules>\\
+\        <module>${schema}</module>\\
+\      </modules>\\
+\    </profile>
 SED_SCRIPT
 fi
 
 
 # Add schema dependency in web/pom.xml
 line=$(grep -n "schema-${schema}" web/pom.xml | cut -d: -f1)
 
-if [ ! $line ]
+if [ ! -n "$line" ]
 then
-  line=$(grep -n 'schema-iso19139</artifactId>' web/pom.xml | cut -d: -f1)
-  insertLine=$(($line + 2))
+  gnSchemasVersion='${project.version}'
+  basedir='${basedir}'
 
-  projectGroupId='${project.groupId}'
-  gnSchemasVersion='${gn.schemas.version}'
-
-  echo "Adding schema ${schema} dependency to web/pom.xml"
+  echo "Adding schema ${schema} dependency to web/pom.xml for schemaCopy"
 
+  line=$(grep -n '<!-- add schema_plugins -->' web/pom.xml | cut -d: -f1 | tail -1)
+  insertLine=$(($line))
   sed $sedopt -f /dev/stdin web/pom.xml << SED_SCRIPT
   ${insertLine} a\\
-\    <dependency>\\
-\      <groupId>${projectGroupId}</groupId>\\
-\      <artifactId>schema-${schema}</artifactId>\\
-\      <version>${gnSchemasVersion}</version>\\
-\    </dependency>
+\        <dependency>\\
+\          <groupId>org.geonetwork-opensource.schemas</groupId>\\
+\          <artifactId>schema-${schema}</artifactId>\\
+\          <version>${gnSchemasVersion}</version>\\
+\        </dependency>
 SED_SCRIPT
-fi
 
+  echo "Adding schema ${schema} dependency to web/pom.xml for schemaUnpack"
 
-# Add schema resources in web/pom.xml
-line=$(grep -n "schemas/${schema}/src/main/plugin</directory>" web/pom.xml | cut -d: -f1)
-
-if [ ! $line ]
-then
-  line=$(grep -n 'schemas/iso19139/src/main/plugin</directory>' web/pom.xml | cut -d: -f1)
-  finalLine=$(($line + 3))
-
-  projectBaseDir='${project.basedir}'
-  baseDir='${basedir}'
-
-  echo "Adding schema ${schema} resources to web/pom.xml"
-
+  line=$(grep -n '</profiles>' web/pom.xml | cut -d: -f1)
+  insertLine=$(($line - 1))
   sed $sedopt -f /dev/stdin web/pom.xml << SED_SCRIPT
-  ${finalLine} a\\
-\                <resource>\\
-\                  <directory>${projectBaseDir}/../schemas/${schema}/src/main/plugin</directory>\\
-\                  <targetPath>${baseDir}/src/main/webapp/WEB-INF/data/config/schema_plugins</targetPath>\\
-\                </resource>
```

Comment:
Previously you can run `mvn process-resources` to copy the files from the schemas directory to `web` project. With this change we only can unpack the schema's zip file. 

Can we split this sed and keep the old behaviour in addition to the new `<profile>`?
Another alternative would be to backpor the code used in `master` branch to copy the schemas  (https://github.com/geonetwork/core-geonetwork/blob/master/web/pom.xml#L1033-L1054)

Reply:
I back ported the code used on master to copy the schemas.